### ISSUE TITLE
Should we support operators (like `+` and `+ reduce`) on atomics?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#16239]
Issues #16237 and #16238 propose convenience features for implicitly reading atomics and supporting special operators on them that map to their methods.  This issue asks whether we should support things like `+` on two atomics or `+ reduce` on an array of atomics.  It may be that these will just fall out if we adopt #16237, or they may require extra work.  Would such support round out our atomic story or be a bridge too far?

**Comment by ronawho** [comment]
I think this will fall out from #16237 and #16238. I'm in favor of both of those, but even if we don't end up implementing them, I'd be in favor of supporting this since it seems like a common case. The main argument against supporting operator overloads for atomics is wanting it to be very visible that atomic operations are happening, but I don't think that applies in the reduction case.

Related to (and possibly duplicates) https://github.com/chapel-lang/chapel/issues/12549

**Comment by gbtitus** [comment]
Agreed, I don't think there's any downside to implementing operators on atomics that will only require reading/loading from them.

**Comment by damianmoz** [comment]
I would add my voice to those of @ronawho and @gbtitus in what you are suggesting for both this and the two other related issues, All of these issues did remind me to look at which is doing Rust on this topic.
